### ISSUE TITLE
Support private module registry

### ISSUE THREAD

**Issue by atikhono** [boltops-tools/terraspace-bundler#6]
## Summary

Please support private Terraform module registry in Terrafile. Terraspace 0.5.10 throws an error:

```
# Terrafile
mod "app-core-reg", source: "terraform-registry.example.com/example/app-core/aws"
# terraspace bundle
ERROR: git clone https://github.com/terraform-registry.example.com/example/app-core/aws 2>&1
```

I'd really love to try out terraspace for my infrastructure, but this issue is blocking me from doing so.

## Motivation

We release modules from a heavy monolith repository to a private registry to speed up terraform init. This seems to be the case for Terraspace as well.

## Unresolved Questions

See similar error when using a private module repo at GitHub:

```
# Terrafile
mod "app-core-git", source: "git@github.com:example-com/app-core.git"
# terraspace bundle
ERROR: git clone https://github.com/git@github.com:example-com/app-core.git 2>&1
```

How do I clone a private repo by SSH? base_clone_url has been changed from git@github.com to https://github.com in 0.3.0. There seems to be no configuration option to set it.

**Comment by tongueroo** [comment]
> RE: https://github.com/git@github.com: bug

Fixed in #8 Released in terraspace-bundler v0.3.3 Will bump the dependencies in the next terraspace version also. Right now, you can manually run `gem install terraspace-bundler` to update.

> I think it could super-nice if the bundler supported the same format for module sources as Terraform: https://www.terraform.io/docs/modules/sources.html. What do you think?

Yep. Think following or getting closer to that same format will be the direction. It's a matter of time. Will consider PRs. No sweat either way, of course. Thanks!

**Comment by guilhermepozo** [comment]
We use Azure Devops Repositories as Enterprise Terraform modules registry. Microsoft azdevops has a Personal Access Token, used as parameter  on git clone proccess (git clone -c http.extraheader="AUTHORIZATION: Basic <TOKEN> or git clone $PAT@repo...).

Terraspace bundle command are asking me Authentication (user and pass) and this is a problem because im need to run it into AZ Devops Pipeline (silently).

Im should to consider this issue will be resolve my problem or im shold create another one issue?

**Comment by tongueroo** [comment]
@atikhono Closed by #12 Most but not all the source types are supported. Mainly the http source is not supported. PRs are welcomed.

Docs: https://terraspace.cloud/docs/terrafile/sources/

@guilhermepozo The `TB_GIT_CLONE_ARGS ` Docs: https://terraspace.cloud/docs/terrafile/env-vars/
